File: RootTools2/src/main/java/com/rarnu/tools/neo/xposed/MIUISearchBar.java
package com.rarnu.tools.neo.xposed;

import de.robv.android.xposed.IXposedHookInitPackageResources;
import de.robv.android.xposed.XSharedPreferences;
import de.robv.android.xposed.callbacks.XC_InitPackageResources;


public class MIUISearchBar implements IXposedHookInitPackageResources {

    public void handleInitPackageResources(XC_InitPackageResources.InitPackageResourcesParam paramInitPackageResourcesParam) throws Throwable {
        XSharedPreferences prefs = new XSharedPreferences(XpStatus.PKGNAME, XpStatus.PREF);
        prefs.makeWorldReadable();
        prefs.reload();
        if (prefs.getBoolean(XpStatus.KEY_REMOVESEARCHBAR, false)) {
            if (paramInitPackageResourcesParam.packageName.equals("com.android.systemui")) {
                paramInitPackageResourcesParam.res.setReplacement(paramInitPackageResourcesParam.packageName, "bool", "config_show_statusbar_search", false);
            }
        }
    }
}

File: RootTools2/src/main/kotlin/com/rarnu/tools/neo/xposed/MIUISearchBar.kt
package com.rarnu.tools.neo.xposed

import com.rarnu.xfunc.XposedRes
import com.rarnu.xfunc.XposedResource
import de.robv.android.xposed.XSharedPreferences

class MIUISearchBar : XposedResource() {
    override fun hook(res: XposedRes) {
        val prefs = XSharedPreferences(XpStatus.PKGNAME, XpStatus.PREF)
        prefs.makeWorldReadable()
        prefs.reload()
        if (res.packageName == "com.android.systemui") {
            if (prefs.getBoolean(XpStatus.KEY_REMOVESEARCHBAR, false)) {
                try {
                    res.res.setReplacement(res.packageName, "bool", "config_show_statusbar_search", false)
                } catch (t: Throwable) {

                }
            }
        }
    }
}
